src/localkiwixserver.cpp

```diff
@@ -84,11 +88,21 @@ void LocalKiwixServer::runOrStopServer()
         auto settingsManager = KiwixApp::instance()->getSettingsManager();
         m_port = ui->PortChooser->text().toInt();
         mp_server->setPort(m_port);
-        m_ipAddress = (ui->IpChooser->currentText() != gt("all")) ? ui->IpChooser->currentText() : "0.0.0.0";
         settingsManager->setKiwixServerPort(m_port);
-        settingsManager->setKiwixServerIpAddress(m_ipAddress);
-        mp_server->setAddress(m_ipAddress.toStdString());
-        m_ipAddress = (m_ipAddress != "0.0.0.0") ? m_ipAddress : QString::fromStdString(kiwix::getBestPublicIp());
+        if (ui->IpChooser->currentText() == gt("all_ips")) {
+            m_ipAddress = QString::fromStdString(kiwix::getBestPublicIp());
+        } else if (ui->IpChooser->currentText() == gt("ipv4")) {
+            mp_server->setIpMode(kiwix::IpMode::IPV4);
+            m_ipAddress = QString::fromStdString(kiwix::getBestPublicIps().addr);
+        } else if (ui->IpChooser->currentText() == gt("ipv6")) {
+            mp_server->setIpMode(kiwix::IpMode::IPV6);
+            m_ipAddress = QString::fromStdString(kiwix::getBestPublicIps().addr6);
+        } else {
+            m_ipAddress = ui->IpChooser->currentText();
+            mp_server->setAddress(m_ipAddress.toStdString());
+        }
```

Comment:
Isn't it better to set `m_ipAddress` from the result of `mp_server->getAddress()` after the server is started?

Reply:
That involves moving the relevant logic of `ui->IpAddress->setText` after the server start right? If we don't have an issue with that, then definitely.